x-pack/plugins/aiops/public/embeddables/change_point_chart/embeddable_change_point_chart_factory.tsx

```diff
@@ -0,0 +1,229 @@
+        {
+          ...timeRangeApi,
+          ...titlesApi,
+          ...changePointControlsApi,
+          getTypeDisplayName: () =>
+            i18n.translate('xpack.aiops.changePointDetection.typeDisplayName', {
+              defaultMessage: 'change point charts',
+            }),
+          isEditingEnabled: () => true,
+          onEdit: async () => {
+            try {
+              const { resolveEmbeddableChangePointUserInput } = await import(
+                './handle_explicit_input'
+              );
+
+              const result = await resolveEmbeddableChangePointUserInput(
+                coreStart,
+                pluginStart,
+                serializeChangePointChartState()
+              );
+
+              changePointControlsApi.updateUserInput(result);
+            } catch (e) {
+              return Promise.reject();
+            }
+          },
+          dataLoading,
+          blockingError,
+          dataViews: dataViews$,
+          serializeState: () => {
+            return {
+              rawState: {
+                timeRange: undefined,
+                ...serializeTitles(),
+                ...serializeTimeRange(),
+                ...serializeChangePointChartState(),
+              },
+              references: [],
+            };
+          },
+        },
+        {
+          ...timeRangeComparators,
+          ...titleComparators,
+          ...changePointControlsComparators,
+        }
+      );
+
+      const ChangePointDetectionComponent = getChangePointDetectionComponent(
+        coreStart,
+        pluginStart
+      );
+
+      return {
+        api,
+        Component: () => {
+          if (!apiHasExecutionContext(parentApi)) {
+            throw new Error('Parent API does not have execution context');
+          }
+
+          const [dataViewId, viewType, fn, metricField, splitField, maxSeriesToPlot, partitions] =
+            useBatchedPublishingSubjects(
+              api.dataViewId,
+              api.viewType,
+              api.fn,
+              api.metricField,
+              api.splitField,
+              api.maxSeriesToPlot,
+              api.partitions
+            );
+
+          const reload$ = useMemo(
+            () =>
+              fetch$(api).pipe(
+                skipWhile((fetchContext) => !fetchContext.isReload),
+                map((fetchContext) => Date.now())
+              ),
+            []
+          );
+
+          const timeRange$ = useMemo(
+            () =>
+              fetch$(api).pipe(
+                map((fetchContext) => fetchContext.timeRange),
+                distinctUntilChanged(fastIsEqual)
+              ),
+            []
+          );
+
+          const lastReloadRequestTime = useObservable(reload$, Date.now());
+          const timeRange = useObservable(timeRange$, undefined);
+
+          let embeddingOrigin;
+          if (apiHasExecutionContext(parentApi)) {
+            embeddingOrigin = parentApi.executionContext.type;
```

Comment:
I'm wondering why you're using the `executionContext` type here instead of the `parentApi.type`?

Reply:
good question, are there any differences/preferences for that?

Reply:
agree. Use `parentApi.type` if you need to identify the parent